pkg/signing/signing.go

```diff
@@ -311,29 +312,81 @@ func Verify(cpath, url string, id uint32, isGroup bool, authToken string, noProm
 			return fmt.Errorf("could not get the signing entity fingerprint: %s", err)
 		}
 
-		// try to verify with local OpenPGP store first
-		signer, err := openpgp.CheckDetachedSignature(elist, bytes.NewBuffer(block.Bytes), block.ArmoredSignature.Body)
+		//fmt.Printf("fingerptinyOOOOOOOOOO: %v\n", fingerprint)
+
+		// remove the local key that signed the container
+		if err := sypgp.RemovePupKey(fingerprint); err != nil {
+			return fmt.Errorf("unable to read keyring: %v", err)
+		}
+
+		// download the new key
+		sylog.Infof("Downloading the new key: %s...", fingerprint[24:])
+		netlist, err := sypgp.FetchPubkey(fingerprint, url, authToken, noPrompt)
 		if err != nil {
-			// verification with local keyring failed, try to fetch from key server
-			sylog.Infof("key missing, searching key server for KeyID: %s...", fingerprint[24:])
-			netlist, err := sypgp.FetchPubkey(fingerprint, url, authToken, noPrompt)
-			if err != nil {
-				return fmt.Errorf("could not fetch public key from server: %s", err)
-			}
-			sylog.Infof("key retrieved successfully!")
+			return fmt.Errorf("could not fetch public key from server: %s", err)
+		}
+		//sylog.Infof("key retrieved successfully!")
+		//fmt.Printf("netList type: %T\n", netlist)
 
-			block, _ := clearsign.Decode(data)
-			if block == nil {
-				return fmt.Errorf("failed to parse signature block")
-			}
+		block, _ = clearsign.Decode(data)
+		if block == nil {
+			return fmt.Errorf("failed to parse signature block")
+		}
 
-			// try verification again with downloaded key
-			signer, err = openpgp.CheckDetachedSignature(netlist, bytes.NewBuffer(block.Bytes), block.ArmoredSignature.Body)
-			if err != nil {
-				return fmt.Errorf("signature verification failed: %s", err)
-			}
+		// store new key
+		sylog.Infof("Storing new key...")
+		if err = sypgp.StorePubKey(netlist[0]); err != nil {
+			return fmt.Errorf("could not store public key: %s", err)
+		}
+
+		// verify the container
+		signer, err := openpgp.CheckDetachedSignature(netlist, bytes.NewBuffer(block.Bytes), block.ArmoredSignature.Body)
+		if err != nil {
+			return fmt.Errorf("signature verification failed: %s", err)
+		}
 
-			if noPrompt {
+//		sylog.Infof("key missing, searching key server for KeyID: %s...", fingerprint[24:])
```

Comment:
Is the commented-out code needed? Can it be removed?

Reply:
Yes, Theres a lot of commented code that can/will be removed! *(wip)*